stream impl

pmlopes opened the pull request:
Needs to be tested and properly reviewed since I'm no scala expert

tsegismont commented:
@pmlopes could you rebase on master? This would restart the PR build with latest setup. Thanks!

tsegismont commented:
@pmlopes Thanks for rebasing. It seems the build fails consistently on one test:
https://travis-ci.org/vert-x3/vertx-mysql-postgresql-client/builds/175978262

Maybe due to the MySQL version?

Narigo commented:
This *may* be due to different handling of dates and times in MySQL and PostgreSQL.

But could you tell me a bit background of what you want to achieve so I can try to help, @pmlopes ?

pmlopes commented:
@Narigo the PR is only an implementation of the streaming interface from sql-common i think the issue we have on travis is related to the version of mysql

Narigo commented on src/main/java/io/vertx/ext/asyncsql/impl/AsyncSQLRowStream.java:
```diff
@@ -0,0 +1,150 @@
+    JsonArray array = new JsonArray();
+    data.foreach(new AbstractFunction1<Object, Void>() {
+      @Override
+      public Void apply(Object value) {
```
As this looks a lot like we have in a different file (checking value and "casting" it) - can they be merged?

Narigo commented on .travis.yml:
```diff
@@ -14,7 +14,7 @@ before_install:
 - sudo /etc/init.d/postgresql stop
 - docker run -e POSTGRES_USER=vertx -e POSTGRES_PASSWORD=password -e POSTGRES_DB=testdb -p 5432:5432 -d postgres
-- docker run -e MYSQL_ALLOW_EMPTY_PASSWORD=yes -e MYSQL_USER=vertx -e MYSQL_PASSWORD=password -e MYSQL_DATABASE=testdb -p 3306:3306 -d mysql
+- docker run -e MYSQL_ROOT_PASSWORD=password -e MYSQL_USER=vertx -e MYSQL_PASSWORD=password -e MYSQL_DATABASE=testdb -p 3306:3306 -d mysql/mysql-server:5.6
```
Is this something that needs to be changed in the README as well? We have `MYSQL_ALLOW_EMPTY_PASSWORD=yes` there

pmlopes replied:
The original readme was wrong and has been fixed

Narigo replied:
👍

pmlopes commented:
@Narigo @tsegismont could you please review. I've updated the base readme which was wrong, moved duplicate code to a single place and added a basic streaming test.

Narigo commented on src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java:
```diff
@@ -658,14 +661,50 @@ protected void checkConsistency(TestContext context, Async async, SQLConnection
+
+            res
+              .handler(row -> {
+                context.assertTrue(Data.NAMES.contains(row.getString(0)));
```
If you use `ORDER BY name` in the SQL statement, it should be possible to use `cnt` to check that the result is correct.

I'd also prefer `count` or something more explicit instead of `cnt` but that's not a blocker to me.
